tests/application_development/code_relocation/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(code_relocation)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources})
+
+# Code relocation feature
+zephyr_code_relocate(src/test_file1.c SRAM2)
+
+zephyr_code_relocate(src/test_file2.c SRAM)
+
+zephyr_code_relocate(src/test_file3.c SRAM2_TEXT)
+zephyr_code_relocate(src/test_file3.c SRAM_DATA)
+zephyr_code_relocate(src/test_file3.c SRAM2_BSS)
+
+zephyr_code_relocate(../../../kernel/sem.c SRAM)
+
+if (CONFIG_RELOCATE_TO_ITCM)
+zephyr_code_relocate(../../../lib/libc/minimal/source/string/string.c ITCM_TEXT)
```

Comment:
should this test depend on the LIBC minimal then?

Reply:
yes, this is the default libc for arm platform.